built_value_generator/lib/src/plugin/checker.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+import 'package:built_value_generator/src/value_source_class.dart';
+
+/// Checks a library for errors related to built_value generation. Returns
+/// the errors and, where possible, corresponding fixes.
+class Checker {
+  Map<AnalysisError, PrioritizedSourceChange> check(
+      LibraryElement libraryElement) {
+    final result = <AnalysisError, PrioritizedSourceChange>{};
+
+    for (final compilationUnit in libraryElement.units) {
+      // Don't analyze generated source; there's nothing to do.
+      if (compilationUnit.source.fullName.endsWith('.g.dart')) continue;
+
+      for (final type in compilationUnit.types) {
+        if (!type.interfaces.any((i) => i.displayName.startsWith('Built')))
+          continue;
+
+        final ValueSourceClass sourceClass = new ValueSourceClass(type);
+        final errors = sourceClass.computeErrors();
+
+        if (errors.isNotEmpty) {
+          final lineInfo = compilationUnit.lineInfo;
+
+          // Report one error on the 'Built' interface. Bundle together all the
```

Comment:
line too long

Reply:
This one's 80 cols which I believe is okay.